clients/spark/core/src/main/scala/io/treeverse/clients/Exporter.scala

```diff
@@ -12,79 +20,162 @@ class Exporter(spark : SparkSession, apiClient: ApiClient, repoName: String, dst
+      p.key as pkey, p.address as paddress, p.etag as petag
+      FROM ${newTableName} n
+      FULL OUTER JOIN ${prevTableName} p
+      ON n.key = p.key
+      WHERE n.etag <> p.etag OR n.etag is null or p.etag is null)
+    """)
+
+    actOnActions(ns, dst, commitID, actionsDF)
+  }
+
+  final val successMsg = "Export completed successfully!"
+  private def actOnActions(ns: String, dst: String, commitID: String, actionsDF: DataFrame): Unit = {
+    if (!export(ns, dst, commitID, actionsDF, false)) {
+      // failed to export files, don't export _success files
+      return
+    }
+    if (!export(ns, dst, commitID, actionsDF, true)){
+      return
+    }
+
+    writeSummaryFile(true, commitID, successMsg)
+  }
+
+
+
+  private def writeSummaryFile(success: Boolean, commitID: String, content : String) = {
+    val suffix = if(success) "SUCCESS" else "FAILURE"
+    val dstPath = resolveURL(new URL(dstRoot), s"EXPORT_${commitID}_${suffix}")
+    val dstFS = dstPath.getFileSystem(spark.sparkContext.hadoopConfiguration)
+
+    val stream = dstFS.create(dstPath)
+    stream.writeChars(content)
+    stream.close()
+  }
+
+  final val prefixLen = 8
+  private def randPrefix() : String = {
+    val gen = Random.alphanumeric.dropWhile(_.isDigit)
+    gen.take(prefixLen).mkString("")
   }
 }
 
 object Exporter {
-  private def handleRow(ns: String, rel: String, serializedConf: SerializableWritable[org.apache.hadoop.conf.Configuration], row: Row): Unit =  {
+  // spark success files must end with "/_SUCCESS", unless they are in the root
+  // dir, then they should be equal to "_SUCCESS".
+  final val sparkSuccessFileRegex = "((.*\\/_SUCCESS)|(^_SUCCESS))$".r
+
+  private def handleRow(ns: String, rootDst: String, serializedConf: SerializableWritable[Configuration], row: Row, successFiles :Boolean): String =  {
     val action = row(0)
-    val key = row(1)
+    val key = row(1).toString()
     val address = row(2).toString()
-    val conf = serializedConf.value
 
-    val srcPath = new org.apache.hadoop.fs.Path( if (address.contains("://")) address else ns + "/" + address)
-    val dstPath = new org.apache.hadoop.fs.Path(rel + "/" + key)
+    key match {
+      case sparkSuccessFileRegex() =>
+        if(!successFiles) {
+          // key match, end gracefully
+          return null
+        }
+      case default =>
+        if (successFiles) {
+          // key doesn't match, end gracefully
+          return null
+        }
+    }
+
+    val conf = serializedConf.value
+    val srcPath = resolveURL(new URL(ns), address)
+    val dstPath = resolveURL(new URL(rootDst), key)
 
     val dstFS = dstPath.getFileSystem(conf)
 
-    action match {
-      case "delete" => {
-        dstFS.delete(dstPath,false) : Unit
+      action match {
+        case "delete" => {
+          try {
+            dstFS.delete(dstPath, false)
+            null
+          } catch {
+            case e : (IOException) =>  s"Unable to delete file ${dstPath.toString}: ${e.toString}"
```

Comment:
Why not catch _any_ `Exception`?  It failed in any such case, no?  (Or is the intent to fail by throwing an exception, and let Spark handle that?)

Reply:
Yes, anything else is unexpected and for all we know should fail Spark entirely.